src/main/java/org/terasology/blockNetwork/ImmutableBlockLocation.java

```diff
@@ -62,10 +63,10 @@ public ImmutableBlockLocation(int x, int y, int z) {
      * of the directionVector to y, and the z-component of the directionVector to z.
      */
     public ImmutableBlockLocation move(Side side) {
-        final Vector3i directionVector = side.getVector3i();
-        return new ImmutableBlockLocation(x + directionVector.x, y + directionVector.y, z + directionVector.z);
+        final Vector3ic directionVector = side.direction();
+        return new ImmutableBlockLocation(x + directionVector.x(), y + directionVector.y(), z + directionVector.z());
     }
```

Comment:
this is equivalent to the following, and I think we don't need a full class for this.

```java
immutablePos.translate(side.getVector3i(), new Vector3i());
```